file/lib/src/sentry_io_overrides_integration.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:sentry/sentry.dart';
+import 'sentry_io_overrides.dart';
+
+/// When installed, every new file will be created as [SentryFile].
+/// When installed, operations will use [SentryFile] instead of dart:io's [File]
+/// implementation whenever [File] is used.
+///
+/// Must be called outside the RunZoneGuardedIntegration.
+/// Otherwise the FileOverrides aren't attached to the root zone.
```

Comment:
Does that mean if there's a custom zone, new files are not `SentryFile` in that Zone even if the integration was installed?

Reply:
I'd interpret it as that they are in the custom zone, but not in the outer root zone. Does that make sense @ueman?

Reply:
@denrase is worth trying out, but I 2nd @ueman 
`global` should affect every zone.

Reply:
Also from the documentation of `IOOverrides.global`:

> These are the [IOOverrides](https://api.flutter.dev/flutter/dart-io/IOOverrides/dart-io/IOOverrides-class.html) that will be used in the root [Zone](https://api.flutter.dev/flutter/dart-io/IOOverrides/dart-async/Zone-class.html), and in [Zone](https://api.flutter.dev/flutter/dart-io/IOOverrides/dart-async/Zone-class.html)'s that do not set [IOOverrides](https://api.flutter.dev/flutter/dart-io/IOOverrides/dart-io/IOOverrides-class.html) and whose ancestors up to the root [Zone](https://api.flutter.dev/flutter/dart-io/IOOverrides/dart-async/Zone-class.html) also do not set [IOOverrides](https://api.flutter.dev/flutter/dart-io/IOOverrides/dart-io/IOOverrides-class.html).

When users run their own zone with IOOverrides, our (global) IOOverride will not apply. This is what we'd expect in that case, maybe something to document?

<img width="676" alt="Bildschirm­foto 2023-04-03 um 12 20 07" src="https://user-images.githubusercontent.com/3984453/229482331-8e4af954-771f-4feb-b363-ad5dc00e91db.png">